central/networkgraph/transformer/external.go

```diff
@@ -0,0 +1,48 @@
+package transformer
+
+import (
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/networkgraph"
+)
+
+type transformExternalDiscoveredImpl struct{}
+
+// Transform will convert any discovered external flows to the Internet entity
+// for masking/anonymization within the network graph.
+func (t *transformExternalDiscoveredImpl) Transform(flows []*storage.NetworkFlow) []*storage.NetworkFlow {
+	ret := make([]*storage.NetworkFlow, 0, len(flows))
+
+	for _, flow := range flows {
+		ret = append(ret, anonymizeDiscoveredFlow(flow))
+	}
+
+	return ret
+}
+
+// Return NetworkEntityInfo_INTERNET if entity is a 'discovered' external entity
+// Otherwise, return entity.
+func anonymizeDiscoveredEntity(entity *storage.NetworkEntityInfo) *storage.NetworkEntityInfo {
+	if networkgraph.IsExternalDiscovered(entity) {
+		return networkgraph.InternetEntity().ToProto()
+	}
+	return entity
+}
+
+// Transform the Src or Dst entities to INTERNET if they are 'discovered' external
+// entities, otherwise the flow is unmodified.
+func anonymizeDiscoveredFlow(flow *storage.NetworkFlow) *storage.NetworkFlow {
+	props := flow.GetProps()
+	src, dst := props.GetSrcEntity(), props.GetDstEntity()
+
+	if !networkgraph.IsExternalDiscovered(src) && !networkgraph.IsExternalDiscovered(dst) {
+		return flow
+	}
+
+	flow = flow.CloneVT()
```

Comment:
I don't think you need `CloneVT` here. That was previously used in the aggregator, because otherwise anonymizing one flow would lead to anonymizing all flows with the same entity. The problem with that is that the loop would skip to the next flow instead of doing the dedpulication if it encountered an anonymized entity. There is no such problem here.